### ISSUE TITLE
UNKNOWN_TO_SDK_VERSION

### ISSUE THREAD

**Issue by kedbirhan** [aws/aws-sdk-java-v2#2343]
there is a problem with RecorderStatus class of AWS config package. API call response capitalizes the RecorderStatus. unfortunately, the SDK matches it to the enum if it's not capitalized. hence returning "UNKNOWN_TO_SDK_VERSION". 
## Expected Behavior

the API response should match the sdk's specification.
## Steps to Reproduce (for bugs)

to reproduce run in the region where aws config is active
 RecorderStatus recorderStatus=configurationRecorderStatus.lastStatus();         System.out.println(configurationRecorderStatus.lastStatus().name().equalsIgnoreCase(RecorderStatus.SUCCESS.name()));
            log.info("{} {}", configurationRecorderStatus.recording().equals(True),**configurationRecorderStatus.lastStatus().equals(RecorderStatus.SUCCESS)**); // if configurationRecorderStatus.lastStatus() is success it should be true but the response is Capitalized so the sdk is unabel to resolve to the defined enum, hence retuns false
           return  configurationRecorderStatus.recording() && configurationRecorderStatus.lastStatus().equals(RecorderStatus.SUCCESS);

## Your Environment

* AWS Java SDK version used: 
* JDK version used:
* Operating System and version: 
java 11
aws sdk for java v2.15.7
mac

**Comment by debora-ito** [comment]
@kedbirhan yes I see the issue, when calling `ConfigClient.describeConfigurationRecorderStatus()` I get `"lastStatus":"SUCCESS"` in the response, and the SDK expects the values in initial uppercase letter only.

The enum is generated based on a model maintained by the service team, I'll contact the AWS Config team and ask for a fix. Thank you for reaching out.

**Comment by debora-ito** [comment]
For visibility to other SDKs, I'm moving this to the `aws-sdk` repository.
